src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -334,8 +338,8 @@ private void checkIfGroupsExists(Course course) {
         User user = userService.getUserWithGroupsAndAuthorities();
         List<Exercise> exercises = exerciseService.findAllForCourse(course, false, user);
 
-        exercises = exercises.stream().filter(exercise -> exercise instanceof TextExercise || exercise instanceof ModelingExercise || exercise instanceof FileUploadExercise)
-                .collect(Collectors.toList());
+        exercises = exercises.stream().filter(exercise -> exercise instanceof TextExercise || exercise instanceof ModelingExercise || exercise instanceof FileUploadExercise
+                || exercise instanceof ProgrammingExercise && exercise.getAssessmentType().equals(AssessmentType.SEMI_AUTOMATIC)).collect(Collectors.toList());
```

Comment:
```suggestion
                || (exercise instanceof ProgrammingExercise && exercise.getAssessmentType().equals(AssessmentType.SEMI_AUTOMATIC))).collect(Collectors.toList());
```

Reply:
I would prefer brackets, even if `&&` is preferred over `||`, but this is absolutely clear now.